scripts/new_upstream_snapshot.py

```diff
@@ -0,0 +1,627 @@
+#!/usr/bin/python3
+# Python 3.6+
+"""Update current branch with commitish.
+For the average upstream snapshot, this is roughly equivalent to:
+
+Run git merge main.
+Remove any cpick* files in debian/patches that exist in main.
+For every other quilt patch run a quilt push and quilt refresh.
+Commit all quilt refresh changes (if any).
+Update changelog accordingly.
+Print release instructions.
+
+The commitish defaults to 'main'
+"""
+
+import argparse
+import subprocess
+from functools import partial
+from pathlib import Path
+from subprocess import CalledProcessError
+from typing import NamedTuple, Optional, Tuple
+
+sh = partial(subprocess.run, check=True, shell=True)
+capture = partial(sh, capture_output=True, universal_newlines=True)
+
+QUILT_COMMAND = "quilt --quiltrc -"
+QUILT_DIFF_ARGS = "-p ab --no-timestamps --no-index"
+QUILT_ENV = {
+    "QUILT_PATCHES": "debian/patches",
+    "QUILT_DIFF_OPTS": "-p",
+    "QUILT_PATCH_OPTS": "--reject-format=unified",
+    "QUILT_DIFF_ARGS": f"{QUILT_DIFF_ARGS} --color=auto",
+    "QUILT_REFRESH_ARGS": f"{QUILT_DIFF_ARGS}",
+    "LANG": "C",
+}
+
+
+class ChangelogDetails(NamedTuple):
+    # This should be cleaner with dataclasses
+    source: str
+    version: str
+    distro: str
+    urgency: str
+    maintainer: str
+    timestamp: str
+    date: str
+    bugs_fixed: str
+    changes: str
+
+    @classmethod
+    def get(cls, offset=0):
+        results = capture(
```

Comment:
This fails ugly if we aren't in a directory or branch w/ debian/changelog.
Pre-flight check?
```suggestion
        if not os.path.exists("debian/changelog"):
            raise RuntimeError(
                "No debian/changelog found. Are we in the right dir/branch?"
            )
        results = capture(
```